helix-core/src/movement.rs

```diff
@@ -245,6 +245,7 @@ impl CharHelpers for Chars<'_> {
 
         // Find our target position(s).
         let head_start = head;
+        #[allow(clippy::while_let_on_iterator)] // Clippy's suggestion to fix doesn't work here.
```

Comment:
I wonder if this should be reported to clippy as false positive?

Reply:
I was planning to do that, but hadn't gotten around to it yet.  Thanks for finding that issue and adding our situation to the list!

Reply:
This should be fixed on next clippy release.